### ISSUE TITLE
Proposal - Allow specifying name and export location of the generated atlas file

### ISSUE THREAD

**Issue by VisenDev** [foxnne/pixi#22]
It would be nice to have some functionality which would allow you to specify where the atlas file is sent to, and what it is called

**Comment by VisenDev** [comment]
> the name you input here will be the name of the exported atlas and image
Yes, I know you can specify where the output folder is, what I meant is using a different folder for the exported png and for the exported JSON data

For example, putting the png in the `images` folder of your game, and the .atlas in `config`

> and what it is called
By this I meant that you could specify a custom name for the exported atlas, such as `textures.json` instead of using `.atlas`

Hope that clarifies

**Comment by foxnne** [comment]
Ah okay! Yes that does. I'll have to think about the best way to go about it, but I think that's doable.

**Comment by foxnne** [comment]
I think @slimsag had a better suggestion for this: we need a config file for a "project". It shouldn't be required for pixi to allow editing files and exporting simple things, but maybe it should be required to use the packing functionality. If a `.pixiproject` file is located in the root folder, it will be read for project specific settings, such as export file names and paths, ldtk compatibility, etc.

**Comment by foxnne** [comment]
Hey! thats awesome! I want to let you know though that things are way out of date now documentation wise, as I have basically completely rewritten the app in the past few months. (this issue is actually fixed in main, I just haven't revisited the issues since as I should have)

I will say however that this project has been how I've been learning zig and if it can be beneficial to you as well, thats great! 

I think for me, the best ways to jump in are to add some new feature, and get to learn the pieces that go in to each.

In the next few days I will revisit the page and get everything up to date so that contributing is easier, and add some good first issues or features that are low priority but would be really nice to have. 

If you don't want to wait on me for that, one thing that comes to mind that would be nice is being able to export the currently selected sprites as images. 

What this would entail is: 

- Add a new keybind, which would be `ctrl/cmd + p` and added [here](https://github.com/foxnne/pixi/blob/30c1f2d2325d2e644f639b7d45db656bd1e3d748/src/editor/Keybinds.zig#L9).
- Process the new keybind from `FileWidget`, [here](https://github.com/foxnne/pixi/blob/30c1f2d2325d2e644f639b7d45db656bd1e3d748/src/editor/widgets/FileWidget.zig#L39) and call function below.
- Add a new function, I suppose `print` to `Editor`, which is kind of a mess of functions right now, but general functions that operate on the active file tend to end up there. Maybe [here](https://github.com/foxnne/pixi/blob/30c1f2d2325d2e644f639b7d45db656bd1e3d748/src/editor/Editor.zig#L763)?
-  Check out `copy` in `Editor`, it grabs each sprite, and copies them into the "transform" layer, which is just essentially writing them into a blank texture/dvui image source, and then crops them down to just the non-transparent pixels. For exporting them with the hotkey, we would essentially just skip the cropping.
- Then we need to offer a dialog to choose the export type and location. As of the latest DVUI, we can now offer exporting png and jpg, so I'd love to support both. This would mean creating a new dialog box gui in pixi, which then opens a native file dialog through DVUI. 
- If possible, the dialog box would let you choose file type through a dropdown, and also export scale, since we would like to export pixel art at sometimes up to 16-32x scale for posting or whatever

Anyway, these are just all individual tasks that could be just adding the keybind, or taking it as far as you'd like. If you aren't interested in this feature, we can find something else you may be interested in working on :) 

Thanks!

**Comment by foxnne** [comment]
That's great to hear! I'm really glad the project isn't too difficult to read and understand. 

I feel like I don't have much to offer in terms of a "vision" for the project, I started this project so I could easily manage drawing and exporting my assets for games I was making for my kids. I haven't really had many users to my knowledge, so this has primarily just been a project where I add features as I need them, and try to keep the UI as simple and clean as possible. DVUI also makes it easy to try to make the app kind of fun with animations for most things. 

So if you ever do want to get into any part of this and have any ideas for features you'd like to support, I'm open to it. 

One example is I was just talking to David about maybe capturing frames at a specified rate into a gif for time lapses of drawing sessions. 

I have future plans of maybe adding Lua and having little games be possible to play or make within the editor itself. 

Just to say, this project can be a lot of things, Id love it to be a modular editor at some point, something where you can author the entire 2d game without leaving the editor.

**Comment by Dlewis89** [comment]
Nice, I had started a python pygame udemy course a year or so ago when i got started just to see if making games was an interest that i wanted to continue to follow but that is the only game development experience i've had.

**Comment by Dlewis89** [comment]
I think i feel confident enough to start tackling an issue now. I'll start on the other issue tomorrow at some point, if you change your mind about what you want me to work on then just let me know.

**Comment by foxnne** [comment]
Upon re-thinking, I guess this issue isn't exactly closed. The project tab in pixi allows you to create a `.pixiproject` file for the base folder, which contains export paths for the combined atlas of that folder. Once this is created, those paths will be loaded each time the project folder is opened.

However, there currently is no mechanism to choose paths without creating a "project" and I think I want to offer the same exact path selection boxes, just remind the user that those paths will not be saved without first creating a project file (which is just a button on the project tab).

**Comment by foxnne** [comment]
Either one would be very beneficial, but I think #78 is probably the more "fun" and larger development if you feel like tackling something a little more involved :)

**Comment by foxnne** [comment]
This was added with `.pixiproject`
